proxyproto.go

```diff
@@ -0,0 +1,251 @@
+package sftp
```

Comment:
Can you add a link to https://github.com/armon/go-proxyproto at the top of this file somewhere so we can track where this code came from? This will be useful if we run into issues or need clarification so we can find the original source.